src/schedulers/dpmsolver_multistep.rs

```diff
@@ -0,0 +1,368 @@
+    /// `algorithm_type: DPMSolverAlgorithmType::DPMSolverPlusPlus`.
+    pub sample_max_value: f32,
+    /// The algorithm type for the solver
+    pub algorithm_type: DPMSolverAlgorithmType,
+    /// The solver type for the second-order solver.
+    pub solver_type: DPMSolverType,
+    /// Whether to use lower-order solvers in the final steps. Only valid for < 15 inference steps. We empirically
+    /// find this can stabilize the sampling of DPM-Solver for `steps < 15`, especially for steps <= 10.
+    pub lower_order_final: bool,
+}
+
+impl Default for DPMSolverMultistepSchedulerConfig {
+    fn default() -> Self {
+        Self {
+            beta_start: 0.0001,
+            beta_end: 0.02,
+            beta_schedule: BetaSchedule::Linear,
+            train_timesteps: 1000,
+            solver_order: 2,
+            prediction_type: PredictionType::Epsilon,
+            sample_max_value: 1.0,
+            algorithm_type: DPMSolverAlgorithmType::DPMSolverPlusPlus,
+            solver_type: DPMSolverType::Midpoint,
+            lower_order_final: true,
+        }
+    }
+}
+
+pub struct DPMSolverMultistepScheduler {
+    alphas_cumprod: Vec<f64>,
+    alpha_t: Vec<f64>,
+    sigma_t: Vec<f64>,
+    lambda_t: Vec<f64>,
+    init_noise_sigma: f64,
+    lower_order_nums: usize,
+    model_outputs: Vec<Tensor>,
+    timesteps: Vec<usize>,
+    pub config: DPMSolverMultistepSchedulerConfig,
+}
+
+impl DPMSolverMultistepScheduler {
+    pub fn new(inference_steps: usize, config: DPMSolverMultistepSchedulerConfig) -> Self {
+        let betas = match config.beta_schedule {
+            BetaSchedule::ScaledLinear => Tensor::linspace(
+                config.beta_start.sqrt(),
+                config.beta_end.sqrt(),
+                config.train_timesteps as i64,
+                kind::FLOAT_CPU,
+            )
+            .square(),
+            BetaSchedule::Linear => Tensor::linspace(
+                config.beta_start,
+                config.beta_end,
+                config.train_timesteps as i64,
+                kind::FLOAT_CPU,
+            ),
+            BetaSchedule::SquaredcosCapV2 => betas_for_alpha_bar(config.train_timesteps, 0.999),
+        };
+        let alphas: Tensor = 1. - betas;
+        let alphas_cumprod = alphas.cumprod(0, Kind::Double);
+
+        let alpha_t = alphas_cumprod.sqrt();
+        let sigma_t = ((1. - &alphas_cumprod) as Tensor).sqrt();
+        let lambda_t = alpha_t.log() - sigma_t.log();
+
+        let step = (config.train_timesteps - 1) as f64 / inference_steps as f64;
+        // np.linspace(0, train_steps - 1, inference_steps + 1).round()[::-1][:-1]
+        let timesteps: Vec<usize> = (0..inference_steps + 1)
+            .map(|i| (i as f64 * step).round() as usize)
+            // discards the 0.0 element
+            .skip(1)
+            .rev()
+            .collect();
+
+        // tch::Tensor doesn't implement `Copy`
```

Comment:
Not sure if that's what you mean here but this is expected of `tch::Tensor`: the `Copy` trait is to be used for values that can be duplicated by just copying bits whereas tensors allocate heap/gpu/... memory and are refcounted on the C++ side.

Reply:
Oh that is a leftover comment, not to forget to mention this in the PR "disclaimers". Anyway the main message was that I am using a for loop instead of, for instance, `vec![Tensore::new(); config.solver_order]` because I can't copy and was asking to you (or anybody interested in contributing) whether it is ok for you (to use a for for init) or if you have a leaner idea :)
I will remove this comment in the next commit.

Reply:
Gotcha, yeah the initialization via a loop sounds reasonable even if it's slightly more verbose than what can be done for `Copy`able objects.

Reply:
k, I will remove that comment in my next commit.